### DISCUSSION TITLE
Doing multiple task while waiting in airflow

### DISCUSSION THREAD

**Discussion by NonlinearNimesh** [apache/airflow#27087] [Q&A]
Hi, I have a scenario where i am uploading a dicom files and one drawback the MR. CT modality have is that we dont know the number of dicoms in one study, so after each dicom upload i am waiting for 30 seconds and see in those 30 seconds  is there any new incoming dicom image, if yes? then don't mark it as complete.

Problem here is i am waiting 30 seconds for each dicom images in the study. which is not a feasible approach when a study like MR and CT has 500 or 1000 dicom images.

So, what i want is while i am waiting for 30 seconds, the dicom images will continue to upload. As upload and sleep are the two different task. To achieve this i have to go back to my upload task which I think is not possible in airflow. so , i was thinking like is there any concept of switching or interrupts in airflow ??

Can any one please help me in this ..... 

Thanks

**Comment by potiuk** [comment]
no idea what dicom is but you should be able to do it with Deferrable Operators. Just look them up in the docs. You need to make a custom Trigger that will provide an Async-Io compatible way to be triggered in asynchronous way when some external event happen - then your custom operator should submit the image, defer itself and the trigger should be written in the way to asynchronously wake up when your upload is complete. That will free the worker to perform another upload.

You can also emulate that with "poke_reschedule' sensors acompanying your operator - that would create more complex DAGs - you wil basically have to have an operator to submit your image and return some unique id as Xcom and then have a sensor that would use the id to query for it's status (and reschedule itself with delay when it determines it's not completed). There ara a number of examples you can find in google and our documentation or stack overflow when you search for those terms.
